src/pudl/metadata/classes.py

```diff
@@ -1748,3 +1762,64 @@ def to_rst(self, top_dir: DirectoryPath, csv_subdir: DirectoryPath, rst_path: st
                 rendered = encoder.to_rst(
                     top_dir=top_dir, csv_subdir=csv_subdir, is_header=header)
                 f.write(rendered)
+
+
+class DatasetteMetadata(Base):
+    """
+    A collection of Data Sources and Resources for metadata export.
+
+    Used to create metadata YAML file to accompany Datasette.
+    """
+
+    data_sources: Dict[str, DataSource]
+    resource_package: Package = Package.from_resource_ids()
+    label_columns: Dict[str, str]
+
+    @classmethod
+    def from_data_source_ids(
+        cls,
+        data_source_ids: Iterable[str] = [
+            'pudl', 'ferc1', 'eia860', 'eia860m', 'eia923'],
+        extra_etl_groups: Iterable[str] = [
+            'entity_eia', 'glue', 'static_eia', 'static'],
```

Comment:
It would be nice if given an `EtlSettings` object (which is built from the one of the yaml settings files) we could identify all of the tables that end up getting materialized, but I don't think that's possible right now (ultimately the SQLAlchemy metadata is built based on the keys associated with the dataframes in the big-ol dataframe dictionary that ends up in the load step.)

Separately, it seems surprising to me here that there's a `static` and a `static_eia` but no `static_ferc1`. I think this is my oversight. It looks like all the resources in the `static` `etl_group` are associated with `ferc1` so it should probably be re-named `static_ferc1`

Reply:
Yeah I think all the `static` is really `static_ferc1`. I can make this change in this PR, or wait to do this on a different `etl_group` PR